Detections/MultipleDataSources/UnusualGuestActivity.yaml

```diff
@@ -0,0 +1,51 @@
+id: acc4c247-aaf7-494b-b5da-17f18863878a
+name: Unusual guest activity - Inviting external guest user followed by Azure AD powershell signin activity
+description: |
+  'By default guests have capability to invite more external guest user, who can do suspicious Azure AD enumeration. This detection will first look at guests 
+  inviting external guests users who are then logging via Azure AD powershell after accpeting invitation.
+  Ref : 'https://danielchronlund.com/2021/11/18/scary-azure-ad-tenant-enumeration-using-regular-b2b-guest-accounts/'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+      - AuditLogs
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+      - SigninLogs
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+  - Persistence
+  - Discovery
+relevantTechniques:
+  - T1078.004
+  - T1136.003
+  - T1087.004
+query: |
+  AuditLogs 
+  | where OperationName in ("Invite external user", "Bulk invite users - started (bulk)","Invite external user with reset invitation status")
+  | extend InitiatedByUser = iff(isnotempty(tostring(parse_json(tostring(InitiatedBy.user)).userPrincipalName)), 
+    tostring(parse_json(tostring(InitiatedBy.user)).userPrincipalName), tostring(parse_json(tostring(InitiatedBy.app)).displayName))
+  | where  InitiatedByUser has_any ("live.com#", "#EXT#")
+  | extend parsedUser = iff(InitiatedByUser has "live.com#", tostring(split(InitiatedByUser, "#")[1]),tostring(split(InitiatedByUser, "#EXT#")[1])) ,  InvitationTime = TimeGenerated
+  | join ( 
+  SigninLogs 
+  | where UserType == "Guest" and AppDisplayName == "Microsoft Azure PowerShell"

```

Comment:
UserType is causing the KQL validation error:
https://dev.azure.com/azure/Azure-Sentinel/_build/results?buildId=32689&view=logs&j=2bbe2297-1b5b-54f3-6060-67e7b289f6c3&t=293f4a23-e36d-5efa-3a4e-ea46d4ab6d83&l=11

`[xUnit.net 00:00:02.97]     Kqlvalidations.Tests.KqlValidationTests.Validate_DetectionQueries_HaveValidKql(detectionsYamlFileName: "UnusualGuestActivity.yaml") [FAIL]
  X Kqlvalidations.Tests.KqlValidationTests.Validate_DetectionQueries_HaveValidKql(detectionsYamlFileName: "UnusualGuestActivity.yaml") [28ms]
  Error Message:
   Template Id:acc4c247-aaf7-494b-b5da-17f18863878a is not valid in Line:9 col:9 Errors:The name 'UserType' does not refer to any known column, table, variable or function., Code: 'KS142', Severity: 'Error', Location: '660..668',The name 'UserType' does not refer to any known column, table, variable or function., Code: 'KS142', Severity: 'Error', Location: '915..923'
Expected: True
Actual:   False
`

Nuget package may need to be updated or backend schema check for SigninLogs needs to be updated, chat with Preeti, Sara and Amit - 
Test run for /home/vsts/work/1/s/.script/tests/KqlvalidationsTests/bin/Release/netcoreapp3.1/Kqlvalidations.Tests.dll(.NETCoreApp,Version=v3.1)

Reply:
Should be fixed in this PR - https://github.com/Azure/Azure-Sentinel/pull/3561